src/equations/NavierStokes_utils.jl

```diff
@@ -0,0 +1,318 @@
+##########################################
+### Via IncompressibleNavierStokes.jl ####
+##########################################
+
+module NavierStokes
+
+using IncompressibleNavierStokes: IncompressibleNavierStokes as INS
+using Lux: Lux
+
+"""
+    create_right_hand_side(setup, psolver)
+
+Create right hand side function f(u, p, t) compatible with
+the OrdinaryDiffEq ODE solvers. Note that `u` has to be an array.
+To convert the tuple `u = (ux, uy)` to an array, use `stack(u)`.
+"""
+create_right_hand_side(setup, psolver) = function right_hand_side(u, p, t)
+    u = eachslice(u; dims = ndims(u))
+    u = (u...,)
+    u = INS.apply_bc_u(u, t, setup)
+    F = INS.momentum(u, nothing, t, setup)
+    F = INS.apply_bc_u(F, t, setup; dudt = true)
+    PF = INS.project(F, setup; psolver)
+    PF = INS.apply_bc_u(PF, t, setup; dudt = true)
+    cat(PF[1], PF[2]; dims = 3)
+end
+
+"""
+    create_right_hand_side_with_closure(setup, psolver, closure, st)
+
+Create right hand side function f(u, p, t) compatible with SciML ODEProblem.
+This formulation was the one proved best in Syver's paper i.e. DCF.
+`u` has to be an array in the NN style e.g. `[n , n, D]`, with boundary conditions padding.
+"""
+create_right_hand_side_with_closure(setup, psolver, closure, st) = function right_hand_side(
+        u, p, t)
+    # not sure if we should keep t as a parameter. t is only necessary for the INS functions when having Dirichlet BCs (time dependent)
```

Comment:
SciML expects functions to have signature `f(u, p, t)` so I would leave it like this even if we do not use it